src/test/java/com/sap/sgs/phosphor/fosstars/data/github/HasSecurityPolicyTest.java

```diff
@@ -1,73 +1,91 @@
 import com.sap.sgs.phosphor.fosstars.tool.github.GitHubProject;
+import com.sap.sgs.phosphor.fosstars.tool.github.GitHubProjectValueCache;
 import java.io.IOException;
 import java.util.Optional;
+import org.junit.AfterClass;
 import org.junit.BeforeClass;
-import org.junit.Ignore;
 import org.junit.Test;
+import org.kohsuke.github.GHContent;
+import org.kohsuke.github.GHRepository;
 import org.kohsuke.github.GitHub;
 
-/**
- * TODO: This test fails when GitHub API rate limit exceeded or some networking issues occur.
- *       The test needs to be re-written to use mocks.
- *       Meanwhile, the test is ignored.
- */
-@Ignore
 public class HasSecurityPolicyTest {
 
-  private static GitHub github;
-
   @BeforeClass
-  public static void initGithubConnection() {
-    try {
-      github = GitHub.connectAnonymously();
-      github.checkApiUrlValidity();
-    } catch (Exception e) {
-      System.out.println("Couldn't connect to GitHub!");
-      e.printStackTrace();
-      github = null;
-    }
+  public static void init() {
+    GitHubDataFetcher.instance().repositoryCache.clear();
+    GitHubDataFetcher.instance().commitsCache.clear();
   }
 
   @Test
-  public void testApacheNifi() throws IOException {
-    assumeNotNull(github);
+  public void testHasPolicy() throws IOException {
+    GHContent content = mock(GHContent.class);
+    when(content.isFile()).thenReturn(true);
+    when(content.getSize()).thenReturn(1000L);
+
+    GHRepository repository = mock(GHRepository.class);
+    when(repository.getFileContent("SECURITY.md")).thenReturn(content);
+
+    GitHub github = mock(GitHub.class);
+    when(github.getRepository(any())).thenReturn(repository);
+
+    GitHubDataFetcher fetcher = GitHubDataFetcher.instance();
+    fetcher.repositoryCache.clear();
+
     HasSecurityPolicy provider = new HasSecurityPolicy(github);
-    ValueHashSet values = new ValueHashSet();
-    GitHubProject project = new GitHubProject("apache", "nifi");
-    provider.set(new Terminal()).update(project, values);
-    assertTrue(values.has(SCANS_FOR_VULNERABLE_DEPENDENCIES));
-    Optional<Value> something = values.of(SCANS_FOR_VULNERABLE_DEPENDENCIES);
-    assertNotNull(something);
-    assertTrue(something.isPresent());
-    Value<Boolean> value = something.get();
-    assertNotNull(value);
-    assertEquals(Boolean.TRUE, value.get());
+    provider.set(new GitHubProjectValueCache());
+
+    check(provider, true);
   }
 
   @Test
-  public void testApacheCxf() throws IOException {
-    assumeNotNull(github);
-    GitHubProject project = new GitHubProject("apache", "cxf");
+  public void testNoPolicy() throws IOException {
+    GHRepository repository = mock(GHRepository.class);
+    when(repository.getFileContent("SECURITY.md")).thenThrow(IOException.class);
+
+    GitHub github = mock(GitHub.class);
+    when(github.getRepository(any())).thenReturn(repository);
+
+    GitHubDataFetcher fetcher = GitHubDataFetcher.instance();
```

Comment:
maybe we can initialize this object in the @Before annotated method?

Reply:
I removed this variable.